tests/e2e/helpers.go

```diff
@@ -0,0 +1,208 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	kerr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/mrogers950/file-integrity-operator/pkg/apis"
+	fileintv1alpha1 "github.com/mrogers950/file-integrity-operator/pkg/apis/fileintegrity/v1alpha1"
+	"github.com/mrogers950/file-integrity-operator/pkg/common"
+)
+
+const (
+	pollInterval         = 2 * time.Second
+	timeout              = time.Second * 60
+	retryInterval        = time.Second * 5
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+)
+
+func cleanUp(t *testing.T, namespace string) func() error {
+	return func() error {
+		f := framework.Global
+
+		if _, err := f.KubeClient.AppsV1().DaemonSets(namespace).Create(cleanAideDaemonset(namespace)); err != nil {
+			return err
+		}
+
+		if err := waitForDaemonSet(daemonSetWasScheduled(f.KubeClient, "aide-clean", namespace)); err != nil {
```

Comment:
Why check for only scheduled and not ready here?

Reply:
because the "clean" DS will never become ready. It finishes, and the daemonset controller detects that the pod died, so it restarts it. The best we can do is figure out if it was scheduled. And since it's a short-running command, it'll most likely be done by then.

Reply:
Ah, that's right.